### PR TITLE
Add player rotation, basic aiming arc

### PR DESCRIPTION
**Pull request by jeffsieu**
Adds an enum `InputMode` to set the current input mode. The game now supports WASD + mouse, as well as controller joysticks for movement and rotating the character.

### CODE REVIEW

File: Scripts/InputMode.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma warning disable SA1514 // Element documentation header should be preceded by blank line
```
Comment (daniellimws) [inline Scripts/InputMode.cs]: still need pragma?

**Comment by daniellimws** [comment]
Shall we remove the 半桶水 xml comments

File: Scripts/InputMode.cs (hunk 1 above)
Comment (jeffsieu) [reply Scripts/InputMode.cs]: idk if not it makes warning ma

Comment (owenl131) [reply Scripts/InputMode.cs]: #pragma siol

**Comment by daniellimws** [comment]
extaransous siol

**Comment by jeffsieu** [comment]
siol

**Review by daniellimws — APPROVED** [review]
I think is ok

**Comment by owenl131** [comment]
So is currently WASD to move and mouse to aim / joystick for everything right?

**Comment by jeffsieu** [comment]
@owenl131 Ya. So we can just add the touch joysticks later to copy the controller joysticks.

**Pull request merged**